config/config.go

```diff
@@ -245,28 +245,103 @@ type BlockBasedAnalyzerConfig struct {
 	// From is the (inclusive) starting block for this analyzer.
 	From int64 `koanf:"from"`
 
+	// If present, the analyzer will run in fast sync mode for
+	// an initial range of blocks, and then switch to slow sync.
+	//
+	// If fast sync mode is enabled:
+	// - DB checks and foreign keys are disabled.
+	// - Multiple analyzers run in parallel and process blocks out of order.
+	// - Analyzers do not perform dead reckoning on state (notably, transfers).
+	// After all analyzers finish the fast sync range:
+	// - DB checks and foreign keys are re-enabled.
+	// - State that would normally be dead-reckoned is fetched directly from
+	//	 the node via the StateToGenesis() RPC.
+	// - A single "slow mode" analyzer (per runtime/consensus) is started, and resumes
+	//   to process blocks sequentially, with dead reckoning enabled.
+	FastSync *FastSyncConfig `koanf:"fast_sync"`
+
 	// To is the (inclusive) ending block for this analyzer.
-	// Omitting this parameter means this analyzer will
-	// continue processing new blocks until the next breaking
-	// upgrade.
+	// Omitting this parameter or setting it to 0 means this analyzer will
+	// continue processing new blocks forever.
 	To int64 `koanf:"to"`
 }
 
-type IntervalBasedAnalyzerConfig struct {
-	// Interval is the time interval at which to run the analyzer.
-	Interval time.Duration `koanf:"interval"`
+type FastSyncConfig struct {
+	// To is the block (inclusive) to which the analyzer
+	// will run in fast sync mode. From this block onwards (exclusive),
+	// the analyzer will run in slow mode. If absent, the analyzer will
+	// only run in slow mode.
+	// Omitting this parameter or setting it to 0 means to use the latest
```

Comment:
This comment about omitting the parameter is not true at the current state of the code, right? Is the plan to implement it that way?

If yes, need to be careful with things like `From: blockCfg.FastSync.To + 1` below in the code.

Reply:
Good catch! An outdated comment. I had this (mostly) implemented but then I decided it wasn't worth the complexity. So I ripped it out, and now the `FastSync.To` parameter has to be a fixed value. I now updated the comment too.